model/src/test/kotlin/ProjectTest.kt

```diff
@@ -34,6 +34,35 @@ private fun readAnalyzerResult(analyzerResultFilename: String): Project =
         .readValue<ProjectAnalyzerResult>().project
 
 class ProjectTest : WordSpec({
+    "Projects" should {
+        "be sorted by type" {
+            val npmProject = Project.EMPTY.copy(id = Identifier.EMPTY.copy(type = "NPM"))
+            val mavenProject = Project.EMPTY.copy(id = Identifier.EMPTY.copy(type = "Maven"))
+
+            val projects = listOf(npmProject, mavenProject)
+
+            projects.sorted() shouldBe listOf(mavenProject, npmProject)
+        }
+    }
+
+    "Projects of the same type" should {
+        "be sorted by definition file depth" {
+            val nestedProject = Project.EMPTY.copy(
+                id = Identifier("Maven", "GroupA", "Artifact", "1.0.0"),
+                definitionFilePath = "nested/pom.xml"
+            )
+
+            val rootProject = Project.EMPTY.copy(
+                id = Identifier("Maven", "GroupB", "Artifact", "1.0.0"),
+                definitionFilePath = "pom.xml"
+            )
+
+            val projects = listOf(nestedProject, rootProject)
```

Comment:
Consider turning `projects` into `sortedProjects` to get AAA sections / blocks.

Reply:
`projects` actually is the *un*sorted list of projects. Do you mean to put the the *actual* or the *expected* list into "sortedProjects"?

Reply:
I've gone with the expected sorter projects now.

Reply:
What I had in mind was something like the following two.
I actually liked your previous change a bit more but I'm ok also as-is.

```
val npmProject = Project.EMPTY.copy(id = Identifier.EMPTY.copy(type = "NPM"))
val mavenProject = Project.EMPTY.copy(id = Identifier.EMPTY.copy(type = "Maven"))

val sortedProjects = listOf(npmProject, mavenProject).sorted()

sortedProjects shouldContainExactly(mavenProject, npmProject)
```

```
val projects = listOf(
  Project.EMPTY.copy(id = Identifier.EMPTY.copy(type = "NPM")),
  Project.EMPTY.copy(id = Identifier.EMPTY.copy(type = "Maven"))
)

val sortedProject = projects.sorted()

sortedProjects shouldContainExactly(mavenProject, npmProject)
```

Reply:
Done.